components/motor/gpio/motor_encoder.go

```diff
@@ -404,52 +401,52 @@ func (m *EncodedMotor) GoFor(ctx context.Context, rpm, revolutions float64, extr
 }
 
 func (m *EncodedMotor) goForInternal(ctx context.Context, rpm, goalPos, direction float64) error {
+	switch speed := math.Abs(rpm); {
+	case speed < 0.1:
+		m.logger.CWarn(ctx, "motor speed is nearly 0 rev_per_min")
+		return motor.NewZeroRPMError()
+	case m.cfg.MaxRPM > 0 && speed > m.cfg.MaxRPM-0.1:
+		m.logger.CWarnf(ctx, "motor speed is nearly the max rev_per_min (%f)", m.cfg.MaxRPM)
+	default:
+	}
+
 	if m.loop == nil {
 		// create new control loop if control config exists
 		if len(m.controlLoopConfig.Blocks) != 0 {
 			if err := m.startControlLoop(); err != nil {
 				return err
 			}
 		} else {
+			// cancel rpmMonitor if it already exists
 			if m.rpmMonitorDone != nil {
 				m.rpmMonitorDone()
 			}
+
+			// start a new rpmMonitor
 			var rpmCtx context.Context
 			rpmCtx, m.rpmMonitorDone = context.WithCancel(context.Background())
 			m.activeBackgroundWorkers.Add(1)
 			utils.ManagedGo(func() {
-				m.rpmMonitor(rpmCtx, rpm, goalPos)
+				m.rpmMonitor(rpmCtx, rpm, goalPos, direction)
 			}, m.activeBackgroundWorkers.Done)
-		}
-	}
 
-	switch speed := math.Abs(rpm); {
-	case speed < 0.1:
-		m.logger.CWarn(ctx, "motor speed is nearly 0 rev_per_min")
-		return motor.NewZeroRPMError()
-	case m.cfg.MaxRPM > 0 && speed > m.cfg.MaxRPM-0.1:
-		m.logger.CWarnf(ctx, "motor speed is nearly the max rev_per_min (%f)", m.cfg.MaxRPM)
-	default:
+			m.stateMu.Lock()
+			defer m.stateMu.Unlock()
+			// start at 10% power
+			if err := m.setPower(ctx, direction*0.1); err != nil {
+				return err
+			}
```

Comment:
tried fully removing this whole section that calls setPower and the motor never starts moving

Reply:
yeah we have to keep it currently, because rpmMonitor waits for a direction before actually updating the power. removing rpmMonitors dependency on [directionMovingInLock ](https://github.com/viamrobotics/rdk/pull/3689/files#diff-c995a39b5b86039be0b4c03ff28ef502f44068848c99b80628eeaa9c9cef734dR251-R253)would let us remove this I think

Reply:
tried this and I'm seeing super weird behavior not really sure what's happening. GoFor is returning early. going to leave it for now but will continue to investigate

Reply:
okay so I think the issue is that `GoFor` has `WaitTilNotPowered` as the return, and if we don't give rpmMonitor an initial power, GoFor can return before any movement has started from rpmMonitor since the motors don't start moving until about 10% power

Reply:
resolving, figured it out by using `WaitForSuccess` in GoFor